graph/telemetry/istio/appender/health.go

```diff
@@ -233,29 +236,29 @@ func (a *HealthAppender) attachHealth(trafficMap graph.TrafficMap, globalInfo *g
 		for namespace, req := range healthReqs {
 			if req.app {
 				wg.Add(1)
-				go func(ctx context.Context, namespace string) {
+				go func(ctx context.Context, namespace, cluster string) {
 					defer wg.Done()
-					h, err := bs.Health.GetNamespaceAppHealth(ctx, business.NamespaceHealthCriteria{Namespace: namespace, IncludeMetrics: false})
+					h, err := bs.Health.GetNamespaceAppHealth(ctx, business.NamespaceHealthCriteria{Namespace: namespace, Cluster: cluster, IncludeMetrics: false})
 					resultsCh <- result{appNSHealth: h, namespace: namespace, err: err}
```

Comment:
Do you also need to add cluster to the `resultsCh`? Based off the comment it's key'd off `namespace+name` so if you had two `details.bookinfo` in two different clusters results from one may override the other.

Reply:
Yeah, good catch, thanks!

Reply:
Yeah, good catch. It is required, and I think I will need to add the cluster in the key too... I need to review this.

Reply:
Or change to something like `map[namespace][cluster][name]`....